GitUI/Editor/FileViewerInternal.cs

```diff
@@ -131,19 +221,21 @@ public void SetText(string text, Action openWithDifftool, bool isDiff = false)
                 _lineNumbersControl.DisplayLineNumFor(text);
             }
 
+            int scrollPos = VScrollPosition;
             TextEditor.Text = text;
 
+            _currentViewPositionCache.Restore(isDiff);
+
+            // Reset the ScrollPos to stop annoying flicker when changing number of visible lines
+            VScrollPosition = scrollPos;
```

Comment:
This should be handled by the `_currentViewPositionCache`.
Have you checked that the functionality of `ViewPositionCache` does _not_ depend on `TextEditor.ShowLineNumbers`?

Reply:
Perhaps the flickering should rather be tackled in a separate PR.

Reply:
I agree. Mixing multiple fixes in a PR makes it harder to trace a source of a regression (if one was introduced)

Reply:
No problems, 
I have removed the syntax highlighting and the fix to the flickering from this branch.
All that remains is the highlighting of all occurrences of the selected text. The others will go into separate PRs.